### PR TITLE
SI-8898 javap -fun under new style lambdas

### PR DESCRIPTION
**Pull request by som-snytt**
To support both -Ydelambdafy strategies, look for both inline
(anonfun) and method (lambda) closure classes.

For method (lambda) style, use the anonfun method that is
invoked by the accessor.

Also, the output of javap must be captured eagerly for
filtering for the current target method.

### CODE REVIEW

**Comment by lrytz** [comment]
PLS REBUILD/pr-scala@a153c05f8a3ea6771d5e0b9afaed4f7e1df00403

**Comment by som-snytt** [comment]
That's a real error; I broke the stripped output.  I'll push the fix shortly. @lrytz 

Actually, just remembered the other issue is that the non-uniformity of the lambda class names means extra work to distinguish class and module closures. That's the other fix to push.

**Comment by som-snytt** [comment]
Dear kitteh,  I guess I am a dog person.

**Comment by som-snytt** [comment]
PLS REBUILD ALL

Pretty please.  On the weekend, like in the old days, when you'd go to the computer room to submit jobs off-hours.

File: test/files/run/repl-javap-memfun.scala [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import scala.tools.partest.JavapTest
+import scala.tools.nsc.Settings
 
+// see repl-javap-lambdas.java for the complementary version
```
Comment (lrytz) [inline test/files/run/repl-javap-memfun.scala]: `.java` :blush:

File: src/repl/scala/tools/nsc/interpreter/package.scala [hunk 2]
```diff
@@ -196,4 +197,13 @@ package object interpreter extends ReplConfig with ReplStrings {
       }
     }
   }
+
+  private[nsc] implicit class `smart stringifier`(val sc: StringContext) extends AnyVal {
```
Comment (lrytz) [inline src/repl/scala/tools/nsc/interpreter/package.scala:202]: is this used somewhere?

File: src/repl/scala/tools/nsc/interpreter/JavapClass.scala [hunk 3]
```diff
@@ -396,7 +409,7 @@ class JavapClass(
     def task(options: Seq[String], classes: Seq[String], inputs: Seq[Input]): Task = {
       //ServiceLoader.load(classOf[javax.tools.DisassemblerTool]).
       //getTask(writer, fileManager, reporter, options.asJava, classes.asJava)
-      import JavaConverters.asJavaIterableConverter
+      //import JavaConverters.asJavaIterableConverter
```
Comment (lrytz) [inline src/repl/scala/tools/nsc/interpreter/JavapClass.scala]: could remove

**Comment by lrytz** [comment]
LGTM - this will require change (again) once closures are emitted in Java-8 style (invokedynamic and LambdaMetaFactory), but it should become easier to find the delambdafy:method-method (no longer need to grep for the `accessor` callsite in the lambda class).

**Comment by som-snytt** [comment]
Pushed the minor edits. Hope it didn't break.

**Comment by som-snytt** [comment]
@lrytz it's an inherently brittle feature, at least as impl'd.  Wasn't sure it was worth it in terms of bang/LOC.  But it's useful when you need it.

**Pull request merged**